core/src/utilities/helpers/routing-helpers.js

```diff
@@ -464,6 +464,34 @@ class RoutingHelpersClass {
     newPath = newPath.replace(/\/$/, '');
     return newPath;
   }
+
+  prepareSearchParamsForClient(currentNode) {
+    let filteredObj = {};
+    if (currentNode && currentNode.clientPermissions && currentNode.clientPermissions.urlParameters) {
```

Comment:
```suggestion
    if (currentNode?.clientPermissions?.urlParameters) {
```
I would suggest to use [Optional chaining (?.)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) which is cleaner.